### PR TITLE
feat: add dropdown item in right column of navigation

### PR DESCRIPTION
**Pull request by niktverd**

### CODE REVIEW

File: src/navigation/components/NavigationDropdownItem/NavigationDropdownItem.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface NavigationDropdownProps {
+    data: NavigationDropdownItem;
+    onClick: MouseEventHandler;
+    isActive: boolean;
+    position: number;
+    hidePopup: () => void;
+}
+
+const NavigationDropdown: React.FC<NavigationDropdownProps> = ({
```
Comment (yuberdysheva) [inline src/navigation/components/NavigationDropdownItem/NavigationDropdownItem.tsx:15]: We agreed that it should be refactored later

**Review by yuberdysheva — CHANGES_REQUESTED** [review]

File: src/navigation/components/NavigationDropdownItem/NavigationDropdownItem.tsx (hunk 1 above)
Comment (niktverd) [reply src/navigation/components/NavigationDropdownItem/NavigationDropdownItem.tsx:15]: ok

File: src/navigation/components/Header/Header.tsx [hunk 2]
```diff
@@ -53,12 +54,16 @@ const MobileMenuButton: React.FC<MobileMenuButtonProps> = ({
 export const Header: React.FC<HeaderProps> = ({data, logo}) => {
     const {leftItems, rightItems} = data;
     const [isSidebarOpened, setIsSidebarOpened] = useState(false);
-    const [activeItemIndex, setActiveItemIndex] = useState(-1);
+    const [activeItemId, setactiveItemId] = useState('');
 
-    const onActiveItemChange = useCallback((index: number) => {
-        setActiveItemIndex(index);
+    const onActiveItemChange = useCallback((index = '') => {
```
[the remark this answers is not in the document]
Comment (niktverd) [reply src/navigation/components/Header/Header.tsx]: sure

File: src/navigation/components/MobileNavigation/MobileNavigation.tsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ const MobileNavigationItem = ({
             e.stopPropagation();
 
             if (onActiveItemChange) {
-                onActiveItemChange(isActive ? '' : `${column}-${index}`);
+                onActiveItemChange(isActive ? undefined : `${column}-${index}`);
```
Comment (yuberdysheva) [inline src/navigation/components/MobileNavigation/MobileNavigation.tsx:73]: isActive && `${column}-${index}` 

?

File: src/navigation/utils.ts [hunk 4]
```diff
@@ -17,5 +17,5 @@ export const getItemClickHandler: ({
     (e) => {
         const id = `${column}-${index}`;
         e.stopPropagation();
-        onActiveItemChange(id === activeItemId ? '' : `${column}-${index}`);
+        onActiveItemChange(id === activeItemId ? undefined : `${column}-${index}`);
```
Comment (yuberdysheva) [inline src/navigation/utils.ts:20]: isActive && ${column}-${index}

?

File: src/navigation/components/MobileNavigation/MobileNavigation.tsx [hunk 5]
```diff
@@ -105,9 +105,9 @@ export interface MobileNavigationProps {
     isOpened?: boolean;
     topItems?: NavigationItemModel[];
     bottomItems?: NavigationItemModel[];
-    activeItemId: string;
+    activeItemId?: string;
     onClose: () => void;
-    onActiveItemChange: (index: string) => void;
+    onActiveItemChange: (index?: string) => void;
```
Comment (yuberdysheva) [inline src/navigation/components/MobileNavigation/MobileNavigation.tsx:110]: id?

File: src/navigation/components/Navigation/Navigation.tsx [hunk 6]
```diff
@@ -14,10 +14,10 @@ const b = block('navigation');
 
 export interface NavigationProps {
     links: NavigationItemModel[];
-    activeItemId: string;
-    onActiveItemChange: (index: string) => void;
+    activeItemId?: string;
     className?: string;
     highlightActiveItem?: boolean;
+    onActiveItemChange: (index?: string) => void;
```
Comment (yuberdysheva) [inline src/navigation/components/Navigation/Navigation.tsx:20]: id?

**Review by yuberdysheva — CHANGES_REQUESTED** [review]

File: src/navigation/components/MobileNavigation/MobileNavigation.tsx (hunk 5 above)
Comment (niktverd) [reply src/navigation/components/MobileNavigation/MobileNavigation.tsx:110]: ok

File: src/navigation/components/Navigation/Navigation.tsx (hunk 6 above)
Comment (niktverd) [reply src/navigation/components/Navigation/Navigation.tsx:20]: ok

File: src/navigation/utils.ts [hunk 7]
```diff
@@ -4,8 +4,8 @@ import {ItemColumnName} from './constants';
 type GetItemClickHandlerArgs = {
     column: ItemColumnName;
     index: number;
-    activeItemId: string;
-    onActiveItemChange: (index: string) => void;
+    activeItemId?: string;
+    onActiveItemChange: (index?: string) => void;
```
[the remark this answers is not in the document]
Comment (niktverd) [reply src/navigation/utils.ts:8]: ok

**Review by yuberdysheva — APPROVED** [review]

**Pull request merged**